test/e2e/tests/test_event_data_store.py

```diff
@@ -0,0 +1,127 @@
+
+CREATE_WAIT_AFTER_SECONDS = 10
+MODIFY_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@pytest.fixture
+def simple_event_data_store():
+    resource_name = random_suffix_name("cloudtrail-eds", 24)
+
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["EVENT_DATA_STORE_NAME"] = resource_name
+
+    resource_data = load_cloudtrail_resource(
+        "event_data_store",
+        additional_replacements=replacements,
+    )
+    logging.debug(resource_data)
+
+    # Create the k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    # Get latest event data store CR
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield (ref, cr)
+
+    # Try to delete, if doesn't already exist
+    try:
+        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+        assert deleted
+    except:
+        pass
+
+@service_marker
+@pytest.mark.canary
+class TestEventDataStore:
+    def test_crud(self, cloudtrail_client, simple_event_data_store):
+        (ref, cr) = simple_event_data_store
+
+        eds_name = cr["spec"]["name"]
+        eds_arn = cr["status"]["ackResourceMetadata"]["arn"]
+
+        cloudtrail_validator = CloudTrailValidator(cloudtrail_client)
+        # verify that data event store exists
+        assert cloudtrail_validator.get_event_data_store(eds_arn) is not None
+
+        # verify that the event data store tags are created
+        eds_tags = cloudtrail_validator.list_resource_tags(eds_arn)
+        tags.assert_ack_system_tags(
+            tags=eds_tags,
+        )
+        tags_dict = tags.to_dict(
+            cr["spec"]["tags"],
+            key_member_name = 'key',
+            value_member_name = 'value'
+        )
+        tags.assert_equal_without_ack_tags(
+            actual = tags_dict,
+            expected = eds_tags,
+        )
+        resource_new_name = f"{eds_name}-renamed"
+
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+        # Update the EDS log file validation
+        cr["spec"]["name"] = resource_new_name
+        k8s.patch_custom_resource(ref, cr)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        eds = cloudtrail_validator.get_event_data_store(eds_arn)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        assert eds["Name"] == resource_new_name
+
+        # Delete k8s resource
+        _, deleted = k8s.delete_custom_resource(ref)
+        assert deleted is True
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        eds = cloudtrail_validator.get_event_data_store(eds_arn)
+        # EventDataStore can't be instantly deleted. They stay on "PENDING DELETION"
+        # state for 7 days before being deleted.
+        assert eds["Status"] == "PENDING_DELETION"
```

Comment:
Consider adding a test case to cover when termination protection is enabled on an EventDataStore; in which case the API endpoint should return a `EventDataStoreTerminationProtectedException`

Also, do we want to consider implementing support for the `RestoreEventDataStore` API endpoint?  https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_RestoreEventDataStore.html

Reply:
> Consider adding a test case to cover when termination protection is enabled on an EventDataStore; in which case the API endpoint should return a `EventDataStoreTerminationProtectedException`

The deletion protection is set to false here: https://github.com/aws-controllers-k8s/cloudtrail-controller/pull/6/files#diff-7d4cb0c6c52934f1aedb71e2adc8a7bc445dfc146e93b54c5179643c95b5783dR7
This is more CloudTrail API behaviour, generally e2e ACK tests should ensure that ACK controllers execute well Read, Create, Update and Delete operations. (and it is not really our responsibility to make sure than an API works well or not).
We have unit tests in runtime library that ensures that a controller will set a Recoverable/Terminal/Advisory condition whenever an API returns an error

>Also, do we want to consider implementing support for the RestoreEventDataStore API endpoint? https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_RestoreEventDataStore.html

Good point. We can consider support this throught a different field like rds did in https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/resource/db_instance/sdk.go#L799-L803
I can address this in a follow up PR.